gene/delete.go

```diff
@@ -339,3 +206,255 @@ func deleteStable(seq *[]dna.Base, delStart int, delEnd int) error {
 	copy((*seq)[delStart:cap(*seq)], (*seq)[delEnd:cap(*seq)]) // shift bases in slice to remove deleted bases
 	return nil
 }
+
+// deleteUpdateGenome updates the genomic sequence upon deletion
+func deleteUpdateGenome(g *Gene, genomeIndexStartPos int, genomeIndexEndPos int) {
+	deletionLen := genomeIndexEndPos - genomeIndexStartPos
+	copy(g.genomeSeq[genomeIndexStartPos:], g.genomeSeq[genomeIndexEndPos:])
+	g.genomeSeq = g.genomeSeq[:len(g.genomeSeq)-deletionLen]
+
+	// Update startPos if necessary
+	if genomeIndexStartPos == 0 {
+		g.startPos += genomeIndexEndPos
+	}
+}
+
+// deleteUpdateCds updates the coding sequence upon deletion
+func deleteUpdateCds(g *Gene, genomeIndexStartPos int, genomeIndexEndPos int) (deletedCodingBases int, err error) {
+	var cdsIndexesToDelete []int
+	var codingDelStart, codingDelEnd int = -1, -1
+	deletionLen := genomeIndexEndPos - genomeIndexStartPos
+	for i := 0; i < len(g.cdsStarts); i++ {
+
+		switch {
+		// if cds is before deletion
+		case genomeIndexStartPos > g.cdsEnds[i]:
```

Comment:
should > be >=?  I am having trouble remembering the numbering system for these variables and I don't see it in the struct declaration.  Is this interval closed-closed or closed-open?  I think we should try to used closed-open if we have control over the struct/file format so that we match golang slices and most numbering in genomics.  I am thinking that the genomeIndex variables are closed-open from the length calc above.